### DISCUSSION TITLE
Shovel not recognizing Source and destination

### DISCUSSION THREAD

**Discussion by paywagh11** [rabbitmq/rabbitmq-server#9440] [Questions]
I have 2(ha) rabbitmq clusters with test.source queue in one rmq and test.target queue on other one.
I tried to create a shovel between these 2 queues with uri in below format:

amqp://username:password@host

In above uri vhost is default '/' so I didn't kept it.  And host is extracted from secret of rmq clusters shown with username password port type.

Could anyone please help me in understanding where I went wrong? Also if my way of getting hostname is ok or not?

**Comment by michaelklishin** [comment]
@PayalMetange we cannot suggest much with such a brief problem statement.

You can fetch URIs from secrets or any other location. URI path is not meant to be omitted, see the [AMQP URI guide](https://rabbitmq.com/uri-spec.html) and Appendix A in particular.

**Reply by paywagh11** [reply]
Hello @michaelklishin . Thanks for the reply. I will be updating my query with screenshot for the issue I faced for source and destination with uri used. Could you please let me know if I can do something to make shovel work.
![IMG-20230917-WA0072~2](https://github.com/rabbitmq/rabbitmq-server/assets/121960364/7645afc6-48e7-47ec-bf6e-49b8cfe8ac8c)

**Reply by michaelklishin** [reply]
See [node logs](https://rabbitmq.com/logging.html#log-file-location) on both sides (and if these are clusters, all nodes on each side) for clues. If this Shovel is in the starting state for minutes then it cannot connect to either source, or destination, or both.

**Comment by michaelklishin** [comment]
Shovels should log a reasonable amount of information about its outbound connections, in particular at debug level.

And RabbitMQ nodes [log a lot of relevant inbound connection information](https://rabbitmq.com/logging.html#logged-events) by default.
